hydra-cluster/src/HydraNode.hs

```diff
@@ -281,6 +281,81 @@ withConfiguredHydraCluster tracer workDir nodeSocket firstNodeId allKeys hydraKe
         hydraScriptsTxId
         (\c -> startNodes (c : clients) rest)
 
+withOfflineHydraNode ::
+  Tracer IO HydraNodeLog ->
+  OfflineConfig ->
+  FilePath ->
+  Int ->
+  SigningKey HydraKey ->
+  (HydraClient -> IO a) ->
+  IO a
+withOfflineHydraNode tracer offlineConfig workDir hydraNodeId hydraSKey action =
+  withLogFile logFilePath $ \logFileHandle -> do
+    withOfflineHydraNode' offlineConfig workDir hydraNodeId hydraSKey (Just logFileHandle) $ do
+      \_stdoutHandle _stderrHandle processHandle -> do
+        result <-
+          race
+            (checkProcessHasNotDied ("hydra-node (" <> show hydraNodeId <> ")") processHandle)
+            (withConnectionToNode tracer hydraNodeId action)
+        case result of
+          Left e -> absurd e
+          Right a -> pure a
+ where
+  logFilePath = workDir </> "logs" </> "hydra-node-" <> show hydraNodeId <.> "log"
+
+withOfflineHydraNode' ::
+  OfflineConfig ->
+  FilePath ->
+  Int ->
+  SigningKey HydraKey ->
+  -- | If given use this as std out.
+  Maybe Handle ->
+  -- -> (HydraClient -> IO a)
+  (Handle -> Handle -> ProcessHandle -> IO a) ->
+  IO a
+withOfflineHydraNode' offlineConfig workDir hydraNodeId hydraSKey mGivenStdOut action =
+  withSystemTempDirectory "hydra-node" $ \dir -> do
+    let cardanoLedgerProtocolParametersFile = dir </> "protocol-parameters.json"
+    readConfigFile "protocol-parameters.json" >>= writeFileBS cardanoLedgerProtocolParametersFile
+    let hydraSigningKey = dir </> (show hydraNodeId <> ".sk")
+    void $ writeFileTextEnvelope (File hydraSigningKey) Nothing hydraSKey
+    let ledgerConfig =
+          CardanoLedgerConfig
+            { cardanoLedgerProtocolParametersFile
+            }
+    let p =
+          -- ( hydraNodeProcess . (\args -> trace ("ARGS DUMP: " <> foldMap (" "<>) (toArgs args)) args) $
+          ( hydraNodeProcess $
+              RunOptions
+                { verbosity = Verbose "HydraNode"
+                , nodeId = NodeId $ show hydraNodeId
+                , host = "127.0.0.1"
+                , -- NOTE(Elaine): port 5000 is used on recent versions of macos
+                  port = fromIntegral $ 5_100 + hydraNodeId
+                , peers
+                , apiHost = "127.0.0.1"
+                , apiPort = fromIntegral $ 4_000 + hydraNodeId
+                , monitoringPort = Just $ fromIntegral $ 6_000 + hydraNodeId
+                , hydraSigningKey
+                , hydraVerificationKeys = []
+                , hydraScriptsTxId = "9fdc525c20bc00d9dfa9d14904b65e01910c0dfe3bb39865523c1e20eaeb0903"
+                , persistenceDir = workDir </> "state-" <> show hydraNodeId
+                , chainConfig = defaultChainConfig
```

Comment:
None of `hydraScriptsTxId` or `chainConfig` (the direct chain one) should be needed here.

In fact, the end-to-end test is currently failing with:

```
hydra-node: user error (FileDoesNotExistError "cardano.sk")
```

Suggesting that we are currently trying to load the `--cardano-signing-key` (via the default chain config). It might have happened only since our rebase, but this confirms to me that we would want to make this situation _impossible_ by using the type system.

---

The error above is only visible after doing some plumbing modifications on this function to actually inherit the stderr like we had done on the `withHydraNode` in the mean-time. Which suggests that it's probably not a good idea to replicate this `withHydraNode` into the `withOfflineHydraNode` in the first place. We would like to pass in the chain config switching the modes instead.